File: manhunt-base/src/main/java/io/github/ytg1234/manhunt/base/config/ManhuntConfig.java
package io.github.ytg1234.manhunt.base.config;

import me.sargunvohra.mcmods.autoconfig1u.ConfigData;
import me.sargunvohra.mcmods.autoconfig1u.annotation.Config;
import me.sargunvohra.mcmods.autoconfig1u.shadowed.blue.endless.jankson.Comment;

import java.util.ArrayList;
import java.util.List;

import static me.sargunvohra.mcmods.autoconfig1u.annotation.ConfigEntry.Gui;

@Config(name = "manhunt")
public class ManhuntConfig implements ConfigData {
    @Gui.EnumHandler(option = Gui.EnumHandler.EnumDisplayOption.BUTTON)
    @Comment(
            "Sets the behaviour of the compass mechanic, can be either UPDATE or USE. UPDATE = Automatically update the compass every tick. USE = Use the compass to update it (more like Dream's manhunt).")
    @Gui.Tooltip(count = 3)
    public Behaviours.Compass compassBehaviour = Behaviours.Compass.USE;

    @Gui.EnumHandler(option = Gui.EnumHandler.EnumDisplayOption.BUTTON)
    @Comment(
            "Sets the behaviour of the damage mechanic. Can be either KILL or DAMAGE. KILL = The speedrunner loses when they are killed (like Dream's manhunt). DAMAGE = The speedrunner loses when they take damage (like dream's assassin).")
    @Gui.Tooltip(count = 3)
    public Behaviours.Damage damageBehaviour = Behaviours.Damage.KILL;

    @Comment("If true, gives players a compass when added to the hunters list.")
    @Gui.Tooltip
    public boolean giveCompassWhenSettingHunters = true;

    // This is a new list because auto config will try to append the values in the Json file.
    @Comment("Dimensions that the compass won't work in.")
    @Gui.Tooltip
    public List<String> disabledDimensions = new ArrayList<>();

    @Comment("Weather to apply the glowing effect to the speedrunner, similar to Dream's Survivalist")
    @Gui.Tooltip
    public boolean highlightSpeedrunner = false;
}

File: manhunt-base/src/main/kotlin/io/github/ytg1234/manhunt/config/ManhuntConfig.kt
package io.github.ytg1234.manhunt.config

import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config
import me.shedaniel.autoconfig.annotation.ConfigEntry
import me.shedaniel.autoconfig.annotation.ConfigEntry.Gui.EnumHandler
import me.shedaniel.cloth.clothconfig.shadowed.blue.endless.jankson.Comment
import java.util.ArrayList

/**
 * A class that holds and represents the configuration of the Manhunt mod.
 *
 * @author YTG1234
 */
@Config(name = "manhunt")
class ManhuntConfig : ConfigData {
    /**
     * Sets the behaviour of the compass mechanic, can be either [UPDATE][Compass.UPDATE] or [USE][Compass.USE].
     *
     * [UPDATE][Compass.UPDATE] = Automatically update the compass every tick.
     * [USE][Compass.USE] = Use the compass to update it (more like Dream's manhunt).
     */
    @EnumHandler(option = EnumHandler.EnumDisplayOption.BUTTON)
    @Comment("Sets the behaviour of the compass mechanic, can be either UPDATE or USE. UPDATE = Automatically update the compass every tick. USE = Use the compass to update it (more like Dream's manhunt).")
    @ConfigEntry.Gui.Tooltip(count = 3)
    @JvmField
    var compassBehaviour = Compass.USE

    /**
     * Sets the behaviour of the damage mechanic. Can be either [KILL][Damage.KILL] or [DAMAGE][Damage.DAMAGE].
     *
     * [KILL][Damage.KILL] = The speedrunner loses when they are killed (like Dream's manhunt).
     * [DAMAGE][Damage.DAMAGE] = The speedrunner loses when they take damage (like dream's assassin).
     */
    @EnumHandler(option = EnumHandler.EnumDisplayOption.BUTTON)
    @Comment("Sets the behaviour of the damage mechanic. Can be either KILL or DAMAGE. KILL = The speedrunner loses when they are killed (like Dream's manhunt). DAMAGE = The speedrunner loses when they take damage (like dream's assassin).")
    @ConfigEntry.Gui.Tooltip(count = 3)
    @JvmField
    var damageBehaviour = Damage.KILL

    /**
     * If true, gives players a compass when added to the hunters list.
     */
    @Comment("If true, gives players a compass when added to the hunters list.")
    @ConfigEntry.Gui.Tooltip
    @JvmField
    var giveCompassWhenSettingHunters = true

    // This is a new list because auto config will try to append the values in the Json file.
    /**
     * Dimensions that the compass won't work in.
     */
    @Comment("Dimensions that the compass won't work in.")
    @ConfigEntry.Gui.Tooltip
    @JvmField
    var disabledDimensions: List<String> = ArrayList()

    /**
     * Whether to apply the glowing effect to the speedrunner, similar to Dream's Survivalist
     */
    @Comment("Whether to apply the glowing effect to the speedrunner, similar to Dream's Survivalist.")
    @ConfigEntry.Gui.Tooltip
    @JvmField
    var highlightSpeedrunner = false
}

